### PR TITLE
first refactor v1.2

### LINKED ISSUE #123: RL-Studio V1.2
Refactoring RL-Studio so it can work:

- multisimulators
- multitask
- multiagent
- multi framework

Changing:

- RL-Studio execution file
- agents/init.py as distribution center to the different tasks, both for training mode and inference
- config.yaml new configuration, with less depth which makes the data more accessible
- agents_type.py
- frameworks_type.py
- tasks_type.py
- envs_type.py
- envs/gazebo/f1/models/init.py

delete main_rlstudio.py

### PR DESCRIPTION
**Pull request by pjfernandecabo**
fix #123 
Refactoring RL-Studio so it can work:

- multisimulators
- multitask
- multiagent
- multi framework

Changing:

- RL-Studio execution file
- agents/init.py as distribution center to the different tasks, both for training mode and inference
- config.yaml new configuration, with less depth which makes the data more accessible
- agents_type.py
- frameworks_type.py
- tasks_type.py
- envs_type.py
- envs/gazebo/f1/models/__init__.py
- delete main_rlstudio.py

### CODE REVIEW

**Comment by rubenlucas93** [comment]
Pending updating the Readme.md inside rl_studio folder.
We don't longer need the cli parameters (except the configuration file path)

**Comment by rubenlucas93** [comment]
Hi @pjfernandecabo I already updated my PRs to be updated with your new one.
Please, review them in order (it is indicated in their description) and don't modify this until those PRS are merged here. Otherwise, I will be forced to align them again to your branch.

Once they are merged you can keep iterating the f1 problems without problem

**Comment by rubenlucas93** [comment]
Hi again @pjfernandecabo !! I assume you already tested your code with the f1 problems, but I don't see the config files updated accordingly.
None of them are working as they are right now. Could you update them so I can set the environment variables and go with them?
Thanks!

**Review by rubenlucas93 — APPROVED** [review]

File: rl_studio/envs/gazebo/f1/models/__init__.py [hunk 1]
```diff
@@ -14,72 +14,124 @@ def __new__(cls, **config):
...
-                DDPGF1FollowLineEnvGazebo,
+            return FollowLaneDQNF1GazeboTF(**environment)
+
+        # =============================
+        # FollowLine - DDPG - TensorFlow
+        # =============================
+        elif (
+            task == TasksType.FOLLOWLINEGAZEBO.value
+            and algorithm == AlgorithmsType.DDPG.value
+            and framework == FrameworksType.TF.value
```
[the remark this answers is not in the document]
Comment (pjfernandecabo) [reply rl_studio/envs/gazebo/f1/models/__init__.py:90]: you control it in the config.yaml file:     

gazebo_environments:
  simple:
    circuit_name: simple

**Pull request merged**